bitbots_vision/bitbots_vision/vision_modules/yoeo_vision_components.py

```diff
@@ -0,0 +1,889 @@
+        self._node: Node = node
+        self._publisher: Optional[rclpy.publisher.Publisher] = None
+
+    def configure(self, config: Dict, yoeo: yoeo_handler.IYOEOHandler) -> None:
+        self._debug_image = DebugImageFactory.create(config)
+        self._field_boundary_detector = YOEOFieldBoundaryDetectorFactory.create(config, yoeo)
+
+        self._register_publisher(config)
+        self._config = config
+
+    def _register_publisher(self, new_config: Dict) -> None:
+        self._publisher = ros_utils.create_or_update_publisher(
+            self._node,
+            self._config,
+            new_config,
+            self._publisher,
+            'ROS_field_boundary_msg_topic',
+            FieldBoundary
+        )
+
+    def run(self, image_msg) -> None:
+        """
+        :param image_msg: Image message
+        :type image_msg:  sensor_msgs.msg._image.Image
+        """
+        convex_field_boundary_points = self._get_convex_field_boundary_points()
+        field_boundary_msg = self._create_field_boundary_msg(image_msg, convex_field_boundary_points)
+        self._publish_field_boundary_msg(field_boundary_msg)
+
+        self._add_field_boundary_to_debug_image()
+        self._add_convex_field_boundary_to_debug_image(convex_field_boundary_points)
+
+    def _get_convex_field_boundary_points(self) -> List[Tuple[int, int]]:
+        return self._field_boundary_detector.get_convex_field_boundary_points()
+
+    @staticmethod
+    def _create_field_boundary_msg(image_msg, field_boundary_points):
+        """
+        :param image_msg: Image message
+        :type image_msg:  sensor_msgs.msg._image.Image
+        :param field_boundary_points: Points of field boundary
+        :type field_boundary_points: List[Tuple[int, int]]
+        :rtype: soccer_vision_2d_msgs.msg._field_boundary.FieldBoundary  # TODO
+        """
+        return ros_utils.build_field_boundary_msg(image_msg.header, field_boundary_points)
+
+    def _publish_field_boundary_msg(self, field_boundary_msg) -> None:
+        """
+        :param field_boundary_msg: Field boundary message
+        :type field_boundary_msg: soccer_vision_2d_msgs.msg._field_boundary.FieldBoundary  # TODO
+        """
+        self._publisher.publish(field_boundary_msg)
+
+    def _add_field_boundary_to_debug_image(self) -> None:
+        field_boundary = self._field_boundary_detector.get_field_boundary_points()
+        self._debug_image.draw_field_boundary(field_boundary, DebugImageColors.field_boundary)
+
+    def _add_convex_field_boundary_to_debug_image(self, convex_field_boundary_points: List[Tuple[int, int]]) -> None:
+        self._debug_image.draw_field_boundary(convex_field_boundary_points, DebugImageColors.field_boundary_convex)
+
+    def set_image(self, image) -> None:
+        """
+        :param image: Image to run the component on
+        :type image:  numpy nd.array (height, width, channels)
+        """
+        self._field_boundary_detector.set_image(image)
```

Comment:
Why do you only set the field_boundary_detector image and not the image for the debug_image drawing? Same for the other set_image calls that have a debug image in their scope.